### ISSUE TITLE
mx_check produces multiple greylisted messages but then skips greylisting

### ISSUE THREAD

**Issue by MuhKuh7** [rspamd/rspamd#2455]
### Classification (Please choose *one* option):

* [ ] Crash/Hang/Data loss
* [ ] WebUI/Usability
* [ ] Serious bug
* [x] Ordinary bug
* [ ] Feature
* [ ] Enhancement

### Reproducibility (Please choose *one* option):

* [ ] Always
* [x] Sometimes
* [ ] Rarely
* [ ] Unable
* [ ] I didn’t try
* [ ] Not applicable

### Rspamd version:
1.7.9
### Operation system, CPU:
Debian 9
### Description (Please provide a descriptive summary of the issue):
Invalid MX result shows multiple greylisting messages but does not greylist in the end.

[MX_INVALID(0.50){greylisted;greylisted;greylisted;greylisted;greylisted;}

This happened with a mail from: https://www.express222.com/email_test.php

### Relevant logs (see details [here](https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd)):
`2018-09-03 22:21:44 #2136(rspamd_proxy) <6eb427>; proxy; proxy_accept_socket: accepted milter connection from 127.0.0.1 port 45328`
`2018-09-03 22:21:45 #2136(rspamd_proxy) <6eb427>; milter; rspamd_milter_process_command: got connection from 184.173.153.213:13727`
`2018-09-03 22:21:45 #2140(normal) <735a65>; task; accept_socket: accepted connection from ::1 port 0, task ptr: 00007F1577CC9F00`
`2018-09-03 22:21:45 #2140(normal) <735a65>; task; rspamd_mime_part_get_cte: detected missing CTE for part as: 7bit`
`2018-09-03 22:21:45 #2140(normal) <735a65>; task; rspamd_message_parse: loaded message; id: <09045f68fcb8c846e8f68f74157ff179@express222.com>; queue-id: <87BA9CC0A6A>; size: 1714; checksum: <959a194232a7e7542e107333968c9a54>`
`2018-09-03 22:21:46 #2140(normal) <735a65>; task; rspamd_mime_text_to_utf8: converted from ISO_8859-1:1987 to UTF-8 inlen: 581, outlen: 581`
`2018-09-03 22:21:46 #2140(normal) <735a65>; task; rspamd_mime_part_detect_language: detected part language: en`
`2018-09-03 22:21:46 #2140(normal) <735a65>; task; fuzzy_generate_commands: <09045f68fcb8c846e8f68f74157ff179@express222.com>, part is shorter than 1000 bytes: 950 (475 * 2.00 bytes), use direct hash`
`2018-09-03 22:21:47 #2140(normal) <735a65>; task; bayes_classify: skip classification as ham class has not enough learns: 0, 200 required`
`2018-09-03 22:21:47 #2140(normal) <735a65>; lua; greylist.lua:258: Score too low - skip greylisting`
`2018-09-03 22:21:47 #2140(normal) <735a65>; task; rspamd_task_write_log: id: <09045f68fcb8c846e8f68f74157ff179@express222.com>, qid: <87BA9CC0A6A>, ip: 184.173.153.213, from: <bounce@legisym.net>, (default: F (no action): [0.59/15.00] [MX_INVALID(0.50){greylisted;greylisted;greylisted;},FORGED_SENDER(0.30){support@legisym.net;bounce@legisym.net;},MIME_HTML_ONLY(0.20){},R_DKIM_ALLOW(-0.20){legisym.net;},R_SPF_ALLOW(-0.20){+ip4:184.173.153.0/24;},IP_SCORE(-0.01){country: US(-0.03);},ARC_NA(0.00){},ASN(0.00){asn:36351, ipnet:184.173.128.0/18, country:US;},DKIM_TRACE(0.00){legisym.net:+;},DMARC_NA(0.00){legisym.net;},FROM_HAS_DN(0.00){},FROM_NEQ_ENVFROM(0.00){support@legisym.net;bounce@legisym.net;},HAS_PHPMAILER_SIG(0.00){},HAS_REPLYTO(0.00){support@legisym.com;},RCPT_COUNT_ONE(0.00){1;},RCVD_COUNT_TWO(0.00){2;},RCVD_TLS_LAST(0.00){},REPLYTO_DOM_NEQ_FROM_DOM(0.00){},TAGGED_FROM(0.00){19b3d6.26e7-###=###.###;},TO_DN_NONE(0.00){},TO_MATCH_ENVRCPT_ALL(0.00){}]), len: 1714, time: 1520.563ms real, 4.502ms virtual, dns req: 28, digest: <959a194232a7e7542e107333968c9a54>, rcpts: <###@###.###>, mime_rcpts: <###@###.###>`
`2018-09-03 22:21:47 #2140(normal) <735a65>; task; rspamd_protocol_http_reply: regexp statistics: 0 pcre regexps scanned, 3 regexps matched, 175 regexps total, 87 regexps cached, 0B bytes scanned using pcre, 2.31k bytes scanned total`

### Expected results:
Only one "greylisted" message and greylisted mail

### Actual results:
Multiple greylisted messages and mail was not greylisted

**Comment by tjharman** [comment]
I can reproduce this.

```
BAYES_HAM (-3) [100.00%]
RWL_HOSTKARMA_WHITE (-1.5) [33.134.84.203.hostkarma.junkemailfilter.com : 127.0.0.1]
MX_INVALID (0.5) [greylisted] <-------------------------------------------------------------- Should be greylisted
DMARC_POLICY_ALLOW (-0.5) [ezibuy.co.nz,none]
R_SPF_ALLOW (-0.2) [+ip4:203.84.134.0/23]
R_DKIM_ALLOW (-0.2) [nz.smxemail.com:s=alpha]
MIME_HTML_ONLY (0.2)
MANY_INVISIBLE_PARTS (0.1) [2]
RCVD_IN_DNSWL_LOW (-0.1) [33.134.84.203.list.dnswl.org : 127.0.3.1]
IP_SCORE (-0.008136) [country: NZ(-0.04)]
RCVD_COUNT_THREE (0) [4]
DKIM_TRACE (0) [nz.smxemail.com:+]
RCVD_TLS_LAST (0)
FROM_EQ_ENVFROM (0)
ASN (0) [asn:38451, ipnet:203.84.134.0/24, country:NZ]
ARC_NA (0)
FROM_HAS_DN (0)
TO_DN_NONE (0)
RCPT_COUNT_ONE (0) [1]
TO_MATCH_ENVRCPT_ALL (0)
```

But it was allowed to pass:

```
Action: No Action
Score:  -4.71 / 15
```

```
root@micro:/usr/share/rspamd/lua# rspamadm configdump greylist
*** Section greylist ***
ipv6_mask = 64;
whitelist_domains_url [
    "/etc/rspamd/local.d/greylist-whitelist-domains.inc",
]
expire = 7776000;
ipv4_mask = 19;
message = "[Greylisted] Go Away";
max_data_len = 10000;
action = "soft reject";
key_prefix = "rg";
timeout = 300;

*** End of section greylist ***


root@micro:/usr/share/rspamd/lua# rspamadm configdump mx_check
*** Section mx_check ***
enabled = true;
greylist_invalid = true;
key_prefix = "rmx";
symbol_good_mx = "MX_GOOD";
symbol_no_mx = "MX_MISSING";
symbol_bad_mx = "MX_INVALID";
timeout = 2;
expire = 5184000;
```

**Comment by tjharman** [comment]
This appears to be fixed in v1.9.0

@MuhKuh7 Do you see the same?

**Comment by tjharman** [comment]
I'm pretty sure this is fixed now stalebot...
